### PR TITLE
support unhex.

### PR DESCRIPTION
**Pull request by hanfei1991**
More details refer to http://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_unhex

@zimulala @shenli @coocood @XuHuaiyu @tiancaiamao PTAL

### CODE REVIEW

**Comment by zimulala** [comment]
check if we should update type inferrer?

**Comment by shenli** [comment]
@hanfei1991 Need to change type infer.

**Comment by XuHuaiyu** [comment]
If we should add a test for this function in evaluator/builtin_string_test.go for not reducing the test coverage?

**Comment by zimulala** [comment]
Add tests in parser_test.go

**Comment by hanfei1991** [comment]
@shenli @zimulala @coocood PTAL

**Comment by shenli** [comment]
@hanfei1991 Still miss test case in evaluator/builtin_string_test.go.

File: executor/executor_test.go [hunk 1]
```diff
@@ -1701,6 +1701,20 @@ func (s *testSuite) TestBuiltin(c *C) {
 	result = tk.MustQuery("select cast('11:11:11' as time)")
 	result.Check(testkit.Rows("11:11:11"))
 
+	// test unhex and hex
+	result = tk.MustQuery("select unhex('4D7953514C')")
```
Comment (shenli) [inline executor/executor_test.go]: It is better to move this test to evaluator package.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

File: evaluator/builtin_string_test.go [hunk 2]
```diff
@@ -684,3 +684,24 @@ func (s *testEvaluatorSuite) TestHexFunc(c *C) {
 
 	}
 }
+func (s *testEvaluatorSuite) TestUnhexFunc(c *C) {
+
```
Comment (zimulala) [inline evaluator/builtin_string_test.go]: Remove this useless line.

**Pull request merged**